app/uk/gov/hmrc/economiccrimelevyreturns/controllers/RelevantAp12MonthsController.scala

```diff
@@ -53,18 +58,76 @@ class RelevantAp12MonthsController @Inject() (
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
     form
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.startAmendUrl))),
-        relevantAp12Months =>
-          eclReturnsConnector
-            .upsertReturn(
-              dataCleanup.cleanup(request.eclReturn.copy(relevantAp12Months = Some(relevantAp12Months)))
+        relevantAp12Months => {
+          val eclReturn = dataCleanup.cleanup(request.eclReturn.copy(relevantAp12Months = Some(relevantAp12Months)))
+          eclReturnsService
+            .upsertReturn(eclReturn)
+            .asResponseError
+            .foldF(
+              error => Future.successful(Status(error.code.statusCode)(Json.toJson(error))),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.